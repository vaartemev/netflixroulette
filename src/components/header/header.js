import React from 'react';
// import { MovieInfo } from '../movieInfo/movieInfo';
import { Logo } from '../logo/logo';
import { SearchField } from '../searchField/searchField';
import './header.scss';

export const Header = () => {
  return (
    <div className="header">
      {/* <MovieInfo /> */}
      <Logo text="netflixroulette"/>
      <SearchField />
    </div>
  );
};
