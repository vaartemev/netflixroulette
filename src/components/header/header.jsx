import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { MovieInfo } from '../movieInfo/movieInfo';
import { Logo } from '../logo/logo';
import { SearchField } from '../searchField/searchField';
import { getMoviesBySearchQuery } from '../../actions';
import './header.scss';

export const Header = () => {
  const searchValue = useSelector(state => state.movie.searchValue);
  const searchFilter = useSelector(state => state.movie.searchBy);
  const sortFilter = useSelector(state => state.movie.sortBy);
  const dispatch = useDispatch();

  const handleOnKeyPress = event => {
    if (event.key === 'Enter') {
      dispatch(getMoviesBySearchQuery(searchValue, searchFilter, sortFilter));
    }
  };

  const movie = useSelector(state => state.movie.movie);
  return (
    <div className="header">
      <Logo text="netflixroulette" />
      {typeof movie !== 'undefined' && 'genres' in movie ? (
        <MovieInfo movie={movie} />
      ) : (
        <SearchField handleOnKeyPress={handleOnKeyPress} />
      )}
    </div>
  );
};
