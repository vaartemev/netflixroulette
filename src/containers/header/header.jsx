import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { MovieInfo } from '../../components/movieInfo/movieInfo';
import { Logo } from '../../components/logo/logo';
import { SearchField } from '../../components/searchField/searchField';
import { getMoviesBySearchQuery } from '../../actions';
import './header.scss';

export const Header = () => {
  const { searchValue, searchFilter, sortFilter } = useSelector(state => ({
    searchValue: state.movie.searchValue,
    searchFilter: state.movie.searchBy,
    sortFilter: state.movie.sortBy
  }));

  const dispatch = useDispatch();

  const handleOnKeyPress = value => {
    dispatch(getMoviesBySearchQuery(value, searchFilter, sortFilter));
  };

  const movie = useSelector(state => state.movie.movie);
  return (
    <div className="header">
      <Logo text="netflixroulette" />
      {typeof movie !== 'undefined' && 'genres' in movie ? (
        <MovieInfo movie={movie} />
      ) : (
        <SearchField handleOnKeyPress={handleOnKeyPress} />
      )}
    </div>
  );
};
