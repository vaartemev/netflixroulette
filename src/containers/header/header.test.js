import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Header } from './header';
import { configureStore } from '../../store';
import { Provider } from 'react-redux';

const store = configureStore();

describe('Header', () => {
  const output = mount(
    <Provider store={store}>
      <Header />
    </Provider>
  );
  it('Render correctly', () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
