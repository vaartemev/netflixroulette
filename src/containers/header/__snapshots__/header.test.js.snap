// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Header>
    <div
      className="header"
    >
      <Logo
        text="netflixroulette"
      >
        <div
          className="logo"
        >
           
          netflixroulette
           
        </div>
      </Logo>
      <SearchField
        handleOnEnterPress={[Function]}
        handleOnInput={[Function]}
        handleOnSearchClick={[Function]}
        searchFilter="title"
        searchValue=""
        sortFilter="release_date"
      >
        <div
          className="search"
        >
          <div
            className="search__title"
          >
            Find your movie
          </div>
          <Input
            handleOnEnterPress={[Function]}
            handleOnInput={[Function]}
            placeholderText="What do you what find?"
            searchValue=""
          >
            <label
              className="search-field__wrapper"
            >
              <input
                className="search__field"
                onInput={[Function]}
                onKeyPress={[Function]}
                placeholder="What do you what find?"
                type="text"
              />
            </label>
          </Input>
          <SearchControls
            handleOnSearchClick={[Function]}
            searchFilter="title"
            searchValue=""
            sortFilter="release_date"
          >
            <div
              className="filters"
            >
              <div
                className="filters__title"
              >
                Search by
              </div>
              <div
                className="filters__buttons"
              >
                <RadioButton
                  filter={[Function]}
                  filters={
                    Array [
                      Object {
                        "key": "title",
                        "title": "Title",
                      },
                      Object {
                        "key": "genres",
                        "title": "Genre",
                      },
                    ]
                  }
                  type="filter"
                >
                  <Button
                    checked={true}
                    handleOnClick={[Function]}
                    key="title"
                    text="Title"
                    type="filter"
                  >
                    <button
                      className="button button__filter-active button__filter"
                      onClick={[Function]}
                      type="button"
                    >
                      Title
                    </button>
                  </Button>
                  <Button
                    checked={false}
                    handleOnClick={[Function]}
                    key="genres"
                    text="Genre"
                    type="filter"
                  >
                    <button
                      className="button button__filter"
                      onClick={[Function]}
                      type="button"
                    >
                      Genre
                    </button>
                  </Button>
                </RadioButton>
              </div>
              <Button
                handleOnClick={[Function]}
                text="Search"
                type="search"
              >
                <button
                  className="button button__search"
                  onClick={[Function]}
                  type="button"
                >
                  Search
                </button>
              </Button>
            </div>
          </SearchControls>
        </div>
      </SearchField>
    </div>
  </Header>
</Provider>
`;
